lib/models/BucketInfo.js

```diff
@@ -137,6 +146,8 @@ class BucketInfo {
         this._replicationConfiguration = replicationConfiguration || null;
         this._cors = cors || null;
         this._lifecycleConfiguration = lifecycleConfiguration || null;
+        // Using undefined because limited to certain features at the moment
+        this._uid = uid || undefined;
```

Comment:
You can generate uid if it doesn't exist. This makes sure that the buckets always get uid old and new.

Reply:
Beware, you then need to make sure all callers of BucketInfo constructor pass the existing uid, otherwise it will generate a new one each time. That was my concern and I though it was safer to just set it to undefined here and let callers set it on a case-by-case basis. Now if we want to make this change everywhere for good reason, why not.

Reply:
For the sake of generating a build today, I think it's best to leave it as is for now (tested and works) and do this change for end of March (if needed).

Reply:
For now, sticking with current implementation as ideas for adding uuid on all buckets should not affect this PR's changes. A follow-up PR will be made soon for testing and adding to a future build